GitCommands/Git/GitModule.cs

```diff
@@ -2470,10 +2469,14 @@ public string GetEffectiveSetting(string setting)
             return EffectiveConfigFile.GetValue(setting);
         }
 
-        public string GetEffectiveGitSetting(string setting, bool cache = true)
+        public EffectiveGitSetting GetEffectiveGitSetting(string setting, bool cache = true)
         {
             GitArgumentBuilder args = new("config") { "--includes", "--get", setting };
-            return GitExecutable.GetOutput(args, cache: cache ? GitCommandCache : null).Trim();
+            ExecutionResult result = GitExecutable.Execute(args, cache: cache ? GitCommandCache : null, throwOnErrorExit: false);
+
+            int resultCode = result.ExitCode ?? 0;
```

Comment:
Do not hide execution errors, should be an error.

Reply:
Wrong. Kind of what I am pointing out. It is NOT a failure. You and @mstv need to understand exit codes and when they are abs not a failure. I detailed in related issue if you will read tied issue.

Reply:
Git config on a config value that is not set still returns a config value.  Bash scripts don't go nuts because value isn't set. We shouldn't either. What I have done is provide not only the config value but a yes I got a value(config key found) Succes status or here your value is blank but you know config value was unset and if we run a set and get an exit code not 0 we have the reason. How about you take the time to read the related issue and understand the change instead of knee jerk reviewing.

Reply:
If there is **no** exit code, it is definitely an error.

Reply:
> If there is **no** exit code, it is definitely an error.

I can change that.  It was just a nullable so had to handle that to make it a value.  Doubt it would ever be null.